include/event2/http.h

```diff
@@ -234,6 +234,25 @@ EVENT2_EXPORT_SYMBOL
 void evhttp_set_default_content_type(struct evhttp *http,
 	const char *content_type);
 
+/**
+ * Set the maximum time in seconds for a connection to live on this server.
+ * A value of zero or less disables the limit.
+ *
+ * An event will fire on any connection that has lived too long that will
+ * close and remove the connection from the server.  This can be used to
+ * mitigate Slowloris DOS attacks which attempt to keep server connections open
+ * by slowly reading or writing data, avoiding connection read/write timeouts,
+ * keeping the connection open for as long as possible.  During a Slowloris
+ * attack, the server will slowly run out of file descriptors or hit it's max
+ * connection limit.  The TTL (Time to Live) limit drops any connections that
+ * have been open for too long.
+ *
+ * @param http the http server on which to set the max connection limit
+ * @param max_ttl the maximum time to live or 0 for no limit.
+ */
+EVENT2_EXPORT_SYMBOL
+void evhttp_set_max_connection_ttl(struct evhttp* http, int max_ttl);
```

Comment:
I think that per-connection bases API also will be helpful, thoughts?

Reply:
I'm not totally sure I'm understanding what you're saying but I think this is a critical feature that should be in the server.  You could add a new connection callback to the server and let the user set their own timeout events but then most people wouldn't know this was necessary and each would implement it in their own way.  To me, this is a feature the server should provide and is analogous to the max header size, max body size and read/write timeouts that are all set on the server but also could have been handled via a new connection callback on a per connection basis.  All that said, I would like to see a new connection callback added.

Reply:
I'm not saying about removing `evhttp_set_max_connection_ttl()` completely, I'm talking about adding `evhttp_connection_set_max_connection_ttl()` that will set this for connection, it will be useful for *outgoing* connections, to specify maximum timeout for the outgoing operations on connection (i.e. send-request/recv-request). And of course for *incoming* connections it should inherit it's value from the server (i.e. the one that installed with `evhttp_set_max_connection_ttl()`)

Reply:
Ah, yes. I've updated the code.